go/vt/vtgate/executor.go

```diff
@@ -1461,6 +1467,30 @@ func (e *Executor) getPlan(vcursor *vcursorImpl, sql string, comments sqlparser.
 	return e.checkThatPlanIsValid(stmt, plan)
 }
 
+func prepareSetVarComment(vcursor *vcursorImpl, stmt sqlparser.Statement) (string, error) {
+	if vcursor == nil || vcursor.Session().InReservedConn() {
+		return "", nil
+	}
+	sysVars := vcursor.Session().GetSystemVariables()
+	if len(sysVars) == 0 {
+		return "", nil
+	}
+	if _, isCommentable := stmt.(sqlparser.Commentable); !isCommentable {
+		vcursor.NeedsReservedConn()
+		return "", nil
+	}
+
+	res := &bytes.Buffer{}
+	for k, val := range sysVars {
+		_, err := res.WriteString(fmt.Sprintf("SET_VAR(%s = %s) ", k, val))
+		if err != nil {
+			return "", vterrors.Errorf(vtrpcpb.Code_INTERNAL, "Buffer exceeded memory limit")
+		}
```

Comment:
```suggestion
    _ = fmt.Fprintf(&res, "SET_VAR(%s = %s) ", k, val)
```

There's no need to check for the error return here because `strings.Builder` cannot return an error. Same for your old code with `WriteString` on `bytes.Buffer`. Those functions cannot error. If they run out of heap they'll panic.